pyartcd/pyartcd/util.py

```
@@ -823,6 +823,28 @@ async def mirror_to_s3(
+
+async def _verify_synced_to_mirrors(source: Union[str, Path], dest: str):
+    """Run a size-only dry-run sync to verify all local files exist at both S3 and R2 destinations."""
+    verify_cmd = ["aws", "s3", "sync", "--no-progress", "--size-only", "--dryrun"]
+    verify_paths = ['--', f'{source}', f'{dest}']
+
+    _, s3_out, _ = await exectools.cmd_gather_async(verify_cmd + verify_paths, env=os.environ.copy())
+    if s3_out.strip():
+        raise IOError(f"S3 post-sync verification failed -- files missing or size mismatch:\n{s3_out.strip()}")
+
+    _, r2_out, _ = await exectools.cmd_gather_async(
+        verify_cmd + ["--profile", "cloudflare", "--endpoint-url", os.environ["CLOUDFLARE_ENDPOINT"]] + verify_paths,
```

Ruff (0.15.14)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)